Raulin0/backup-juggler
# Feature/backup funcionality

## Description
This pull request introduces significant enhancements to the project, including the addition of a backup functionality and refactoring of existing code.

## Changes Made

1. `backup_juggler/juggler.py`:

- Added `JugglerModel` class to represent the data model for the backup operation.
- Implemented `JugglerView` class to handle the view and progress bar for the backup operation.
- Created `JugglerController` class to handle the control logic for the backup operation.

2. `backup_juggler/juggler_cli.py`:

- Implemented backup function to perform the actual backup operation from a source path to a destination path.
- Improved the command-line help message and provided usage examples.

## Motivation and Context
The addition of the backup functionality and the refactoring of code are motivated by the need for a reliable and efficient backup solution. These changes provide a comprehensive backup feature that can handle multiple sources and destinations concurrently. The refactoring improves the code's structure and readability, making it easier to maintain and enhance in the future.

## How to Test

1. Ensure that the project dependencies, including `tqdm`, are installed.
2. Run the test suite to verify the existing functionality.
3. Perform backup operations with various file types (e.g., text files, images) and check if the files are copied successfully.
4. Verify that the progress bar is displayed and updated accurately during the file copying process.
5. Ensure that the original functionality of the backup_juggler module remains intact.

## Pull Request Check List
